src/dotnet-roslyn-tools/PRTagger/PRTagger.cs

```diff
@@ -2,133 +2,186 @@
             }
-
-            // Find PRs between product commit SHAs
-            var prDescription = new StringBuilder();
-            var isSuccess = PRFinder.PRFinder.FindPRs(previousProductCommitSha, currentProductCommitSha, PRFinder.PRFinder.DefaultFormat, logger, gitHubRepoPath, prDescription);
-            if (isSuccess != 0)
+            else
             {
-                // Error occurred; should be logged in FindPRs method
-                continue;
+                logger.LogInformation($"Repo already exists at {gitHubRepoPath}");
             }
+        }
+        catch (Exception ex)
+        {
+            logger.LogError($"Exception while cloning repo: " + ex);
+            return TagResult.Failed;
+        }
 
-            logger.LogInformation($"Creating issue...");
+        // Find PRs between product commit SHAs
+        var prDescription = new StringBuilder();
+        var isSuccess = PRFinder.PRFinder.FindPRs(previousProductCommitSha, currentProductCommitSha, PRFinder.PRFinder.DefaultFormat, logger, gitHubRepoPath, prDescription);
+        if (isSuccess != 0)
+        {
+            // Error occurred; should be logged in FindPRs method
+            return TagResult.Failed;
+        }
 
-            // Create issue
-            await TryCreateIssueAsync(gitHubRepoName, vsBuild, prDescription.ToString(), settings.GitHubToken, logger).ConfigureAwait(false);
+        var issueTitle = $"[Automated] PRs inserted in VS build {vsBuild}";
+        var hasIssueAlreadyCreated = await HasIssueAlreadyCreatedAsync(gitHubClient, gitHubRepoName, issueTitle, logger).ConfigureAwait(false);
+        if (hasIssueAlreadyCreated)
+        {
+            logger.LogInformation($"Issue with name: {issueTitle} exists in repo: {gitHubRepoName}. Skip creation.");
+            return TagResult.IssueAlreadyCreated;
         }
 
-        return 0;
+        logger.LogInformation($"Creating issue...");
+
+        // Create issue
+        return await TryCreateIssueAsync(gitHubClient, issueTitle, gitHubRepoName, prDescription.ToString(), logger).ConfigureAwait(false);
+    }
+
+    public static async Task<ImmutableArray<(string vsBuild, string vsCommit, string previousVsCommitSha)>> GetVSBuildsAndCommitsAsync(
+        AzDOConnection devdivConnection,
+        ILogger logger,
+        int vsBuildNumber,
+        CancellationToken cancellationToken)
+    {
+        var builds = await devdivConnection.TryGetBuildsAsync(
+            "DD-CB-TestSignVS",
+            logger: logger,
+            maxBuildNumberFetch: vsBuildNumber,
+            resultsFilter: BuildResult.Succeeded,
+            buildQueryOrder: BuildQueryOrder.FinishTimeDescending).ConfigureAwait(false);
+        var vsRepository = await GetVSRepositoryAsync(devdivConnection.GitClient);
+        if (builds is not null)
+        {
+            // Find previous VS commit SHA
+            var buildInfoTask = builds.Select(async build =>
```

Comment:
Do we need to check if the build is for the main branch?

Reply:
I don't think so, if it merged into other branch, it should still work.